### ISSUE TITLE
Wrong sorting of products in taxon after saving

### ISSUE THREAD

**Issue by delyriand** [Sylius/Sylius#15954]
**Sylius version affected**: 1.2.11 and higher

**Description**  

The position of the last products in a taxon is not correctly saved when there are more products than the taxon of the 1st productTaxon.

Because the `getMaxPosition` method in `Sylius\Bundle\CoreBundle\Controller\ProductTaxonController` class returns the wrong maximum position.
This is because when we load the "productTaxon" via the line `$productTaxon = $this->findOr404($configuration);` we have no criteria. So the `SingleResourceProvider` does a `findOneBy` with an empty array and returns the 1st "productTaxon".

**Steps to reproduce**  

- with the default fixutres
- try to reorder all products in the Jeans taxon `/admin/products/taxon/9`
- position of 2 last product is "-1"

![image](https://github.com/Sylius/Sylius/assets/465524/fac0714d-cfa0-4202-b6d1-2b8772d64a1f)

**Possible Solution**  

We could define the criteria in the `requestConfiguration` before calling the `getMaxPosition` method. For example:

```php
$firstProductTaxonId = array_key_first($productTaxonsPositions);
$configuration->getParameters()->set('criteria', ['id' => $firstProductTaxonId]);
$maxPosition = $this->getMaxPosition($configuration);
```

**Comment by jakubtobiasz** [comment]
Hi @delyriand,
can you tell use what values did you provide? We aren't able to reproduce your issue. It'd be great if you could record your screen 🙏🏻.

**Comment by delyriand** [comment]
Hi @jakubtobiasz,

I can't reproduce the -1 positions… But if you look at the "maxPosition" (via xdebug or similar) when saving product positions in Jeans taxon, you'll see that it's 5 but not 7.

And if you try to save these positions:

![image](https://github.com/Sylius/Sylius/assets/465524/6edba491-c309-4f6c-a934-86cea349d8b8)

I should have:

- 666F
- 111F
- 000F
- …

Products are saved in the wrong order:

![image](https://github.com/Sylius/Sylius/assets/465524/0e1553b1-5d75-4d97-891a-22aaf8ffc770)

Does it help you reproduce my issue?

**Comment by TheMilek** [comment]
Hi @delyriand!

Thank you for raising the issue! 🏅 
I checked it again after your recent input and I must admit that it doesn't seem to be working properly. 🤔

**Comment by delyriand** [comment]
Hi @TheMilek!

I think you can close the issue and the related PR, as @GSadee has merged another fix from PR #16477
